File: PDOO/p4/civitas_rb/lib/sorpresa_salircarcel.rb
module Civitas
  
class Sorpresa_salircarcel < Sorpresa
  
  def initialize(mazo)
    super("Quedas libre de la carcel")
    @mazo = mazo
  end
  
  
  def aplicar_a_jugador(actual, todos)
    
    if (jugador_correcto(actual, todos))
      informe(actual, todos)
      poseido = false
      
      for i in 0...todos.length
        
        if todos.at(i).tiene_salvoconducto
          poseido = true
        end
      end
      
      if !poseido
        todos.at(actual).obtener_salvoconducto(self)
        salirdelmazo
      end
    end
  end
  
  def salirdelmazo
    @mazo.inhabilitar_carta_especial(self)
  end
  
  def usada
    @mazo.habilitar_carta_especial(self)
  end
  
  def to_string()
    return super
  end
  
end
end

File: PDOO/p4/Civitas/src/civitas/SorpresaSalirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author aida
 */
public class SorpresaSalirCarcel extends Sorpresa {
    
    private MazoSorpresas mazo;
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            Boolean poseido = false;
            for (int i = 0; i < todos.size(); i++){
                if (todos.get(i).tieneSalvoconducto() == true){
                    poseido = true;
                }
            }
            if (poseido = false){
                todos.get(actual).obtenerSalvoconducto(this);
                salirDelMazo();
            }
        }
    }
    
    void salirDelMazo(){
        mazo.inhabilitarCartaEspecial(this);
    }
    
    SorpresaSalirCarcel(MazoSorpresas mazo){
        //Para construir la sorpresa que permite evitar la carcel
        super("Quedas libre de la carcel");
        this.mazo = mazo;
    }
    
    void usada(){
        mazo.habilitarCartaEspecial(this);
    }
    
    @Override
    public String toString(){
        return "Sorpresa{" + "texto=" + super.getTexto() + "}";
    }
}
